chain/rosetta-rpc/src/adapters/mod.rs

```diff
@@ -144,79 +146,313 @@ pub(crate) async fn convert_block_to_transactions(
+            block_hash: &block.header.hash,
+            factory: Some(TransactionsFactory {
+                view_client_addr,
+                block,
+                signers: None,
+                polling_config,
+            }),
+        }
+    }
+
+    /// Creates a new converter for creating list of transactions in specified
+    /// block except that no related transactions will be included.
+    #[cfg(test)]
+    fn new_for_tests(hash: &'a near_primitives::hash::CryptoHash) -> Self {
+        Self { map: std::collections::HashMap::new(), block_hash: hash, factory: None }
+    }
+
+    /// Returns Rosetta transactions mapping to given account state changes.
+    async fn convert(
+        mut self,
+        runtime_config: &near_primitives::runtime::config::RuntimeConfig,
+        accounts_changes: near_primitives::views::StateChangesView,
+        accounts_previous_state: std::collections::HashMap<
+            near_primitives::types::AccountId,
+            near_primitives::views::AccountView,
+        >,
+    ) -> crate::errors::Result<std::collections::HashMap<String, crate::models::Transaction>> {
+        convert_block_changes_to_transactions(
+            &mut self,
+            runtime_config,
+            accounts_changes,
+            accounts_previous_state,
+        )
+        .await?;
+        Ok(self.map)
+    }
+
+    /// Returns a Rosetta transaction object for given state change cause.
+    ///
+    /// `transaction_identifier`, `related_transactions` and `metadata` of the
+    /// object will be populated but initially the `operations` will be an empty
+    /// vector.  It’s caller’s responsibility to fill it out as required.
+    async fn get_for_cause(
+        &mut self,
+        cause: near_primitives::views::StateChangeCauseView,
+    ) -> crate::errors::Result<&mut crate::models::Transaction> {
+        use std::collections::hash_map::Entry;
+        let (identifier, tx_hash) = convert_cause_to_transaction_id(&self.block_hash, cause)?;
+        Ok(match self.map.entry(identifier.hash) {
+            Entry::Occupied(entry) => entry.into_mut(),
+            Entry::Vacant(entry) => {
+                let tx = Self::make_transaction(&mut self.factory, entry.key(), tx_hash).await?;
+                entry.insert(tx)
+            }
+        })
+    }
+
+    /// Constructs a new Rosetta transaction object.
+    ///
+    /// If `tx_hash` is not None, fills out transaction’s `related_transactions`
+    /// with receipts for Near transaction with given hash.
+    async fn make_transaction(
+        factory: &mut Option<TransactionsFactory<'a>>,
+        hash: &String,
+        tx_hash: Option<near_primitives::hash::CryptoHash>,
+    ) -> crate::errors::Result<crate::models::Transaction> {
+        let id = crate::models::TransactionIdentifier { hash: hash.clone() };
+        let related_transactions = match (factory, tx_hash) {
+            (Some(factory), Some(tx_hash)) => factory.get_related_transactions(tx_hash).await?,
```

Comment:
Well, receipts should also have `related_transactions` since we want to be able to tie all the children receipts to the parent transaction